### PR TITLE
basic root dir pre check on fs storage.New

### LINKED ISSUE #59: In disk storage mode, do initial check that disk storage root exists

### PR DESCRIPTION
**Pull request by robjloranger**
`New()` checks for root directory present in fs, returns an error if it does not exist or if any other problem was encountered.

closes #59

### CODE REVIEW

**Comment by robjloranger** [comment]
I forgot to check for existing usage in Olympus or Zeus so don't merge yet

**Comment by robjloranger** [comment]
ok I found all uses of the implementation and updated them

File: pkg/storage/fs/all_test.go [hunk 1]
```diff
@@ -37,7 +37,10 @@ func (d *FsTests) SetupTest() {
 	memFs := afero.NewOsFs()
 	r, err := afero.TempDir(memFs, "", "athens-fs-tests")
 	d.Require().NoError(err)
-	d.storage = NewStorage(r, memFs)
+	d.storage, err = NewStorage(r, memFs)
+	if err != nil {
```
Comment (arschles) [inline pkg/storage/fs/all_test.go]: can you use `d.Require().NoError(err)` here?

**Comment by arschles** [comment]
@robjloranger I left one comment but forgot to "request changes"  in my review. can you make that one change, and then I'll merge?

**Comment by robjloranger** [comment]
All set @arschles

**Pull request merged**